### DISCUSSION TITLE
New user interface for the main list

### DISCUSSION THREAD

**Discussion by majkinetor** [1Remote/1Remote#419] [Ideas]
Column based interface is not particularly suited for all use cases. I propose something more like this:

![image](https://github.com/1Remote/1Remote/assets/85767/442dfbfa-b218-412d-97a8-baf6eaabbdbf)

Red tags are system one, blue are user ones. Protocol should be a system tag anyway, and tags should not be in column as much as Notes (or any bigger chunk of text), which leaves only Name and Target. So we should go for a list based interface which will open up wasted space with column separators.

The benefits of this layout are:

1. You can add easily more items you can have on connections in the future
2. You no longer have weird columns like Note which are not suited for the task
3. You can have more views easily - like we have now card view, we could add in the future something more akin to how Explorer shows folders: detailed view, concize view, card view, or even table view that we have now eventually (since now its only one of the alternatives so who cares how it looks, if you want columns you will have them). I certainly wouldn't do that now (or ever) but its a possibility for future enhacement.

**Comment by majkinetor** [comment]
From discussion on #416:

> @VShawn: It seems that each connection has a different height, which appears to affect the virtualization of the WPF listview. I'm not sure, as I've mentioned I'm not a UI expert.

> @majkinetor: OK, so maybe we ditch WPF and use HTML instead? In HTML all this is trivial.
>> @VShawn Electron has poor performance, I don't like it. a small feature takes 100MB of space and even more memory.
>>> @majkinetor: For sure. I wasn't thinking about electron at all, more something like [Ultralight](https://ultralig.ht) - only 10mb. Not saying it should be that one, or [RmlUI](https://github.com/mikke89/RmlUi) or [Sciter](https://sciter.com/) (5MB) but this is just on top of my head, I didn't research this recently.

> @VShawn : ..working on the UI takes too much time, and there is no sense of accomplishment :(
>> @majkinetor: If I mange to make something on my own it I will publish it in specific branch.

**Comment by VShawn** [comment]
Variable height needs to be treated with caution, not only because of virtualization issues, but also because occupying too much height means that users will see fewer connections.

**Reply by majkinetor** [reply]
Nah, its really simple. We will have following options:

1. Do not show note 
2. Show first line of the note
3. Show entire note

That should be either in global settings or toggable ad-hoc (for example via part of the predefined view, point 3 in original post).

Any system must provide options, and guide the users via defaults, but not block them to use their own ways...  In this case, there are times when you need markdown notes in full (for example you present infrastructure to the team members) and times when you don't (every day work with the same set of servers). But there may be those who will propagate important messages to its team all the time...

**Comment by majkinetor** [comment]
@VShawn have you maybe checked Blazor Desktop?

There is a cross-platform sample here: https://github.com/edandersen/blazor-desktop-crossplatform-sample

**Reply by VShawn** [reply]
Nope, I don't have plans to make 1REMOTE cross-platform now. 

mainly because:
1. I'm not a Linux desktop user, and I have limited experience with SSH and the intricacies of Linux systems.
2. Rewriting the RDP portion for cross-platform compatibility would require double the development and testing efforts. The current RDP library only supports Windows.

**Reply by majkinetor** [reply]
I know, I follow :)

The point was not about it, but about a capable general UI library. The fact Blazor is cross-platform is only a bonus that might be needed one day.

**Reply by VShawn** [reply]
I make habit to check the progress of Blazor and MAUI every few months. Based on the information so far, it seems that Blazor is still a WIP and may not be as popular as other frameworks at the moment.

https://www.reddit.com/r/dotnet/comments/15cq4xv/unpopular_opinion_blazor_is_only_creating_a/
https://www.reddit.com/r/Blazor/comments/vaccxa/hot_reload_is_rubbish/
https://www.reddit.com/r/dotnet/comments/12fin5d/net_desktop_app/

**Reply by majkinetor** [reply]
I looked into all those and I get a different feeling about Blazor Desktop. It looks like people generally love it.

**Reply by VShawn** [reply]
Oh? I'll go check it out.
